src/lua/httpc.lua

```diff
@@ -54,7 +64,10 @@ local http_new = function(opts)
     opts.max_total_connections = opts.max_total_connections or 0
 
     local curl = driver.new(opts.max_connections, opts.max_total_connections)
-    return setmetatable({ curl = curl, }, curl_mt )
+    return setmetatable({
+        curl = curl,
+        encoders = table.deepcopy(encoders),
```

Comment:
Nit: `table.copy()` is enough, it is one level table (the same for the default instance below).